src/Annotations/Annotation.cpp

```diff
@@ -95,32 +101,40 @@ handling for it is here just in case
 TOOL_CHECKER Calls target = FUNCTION(does_free).PARAM(1) methods = free
 -- ParameterAnnotation
 
-TOOL_CHECKER Calls target = FUNCTION(creates_obligation).PARAM(2).FIELD(x)
+TOOL_CHECKER Calls target = FUNCTION(creates_obligation).PARAM(2).FIELD(0)
 methods = free
 -- ParameterAnnotation with field
 
 TOOL_CHECKER MustCall target = FUNCTION(creates_obligation).RETURN methods =
 free
 -- ReturnAnnotation
 
-TOOL_CHECKER Calls target = FUNCTION(does_something).RETURN.FIELD(x) methods =
+TOOL_CHECKER Calls target = FUNCTION(does_something).RETURN.FIELD(0) methods =
 free
 -- ReturnAnnotation with field
 
-TOOL_CHECKER MustCall target = STRUCT(my_struct).FIELD(x) methods = free
+TOOL_CHECKER MustCall target = STRUCT(my_struct).FIELD(0) methods = free
 -- StructAnnotation; always has FIELD specifier
 
 Example for chunks:
 
-TOOL_CHECKER Calls target = FUNCTION(creates_obligation).PARAM(2).FIELD(x)
-methods = free1, free2 chunks = ['TOOL_CHECKER', 'Calls', 'target', '=',
+TOOL_CHECKER Calls target = FUNCTION(creates_obligation).PARAM(2).FIELD(0)
+methods = free1, free2
+
+chunks = ['TOOL_CHECKER', 'Calls', 'target', '=',
 'FUNCTION(creates_obligation).PARAM(2).FIELD(x)', 'methods', '=', 'free1,',
-'free2'] chunks[0] should be TOOL_CHECKER to help verify it's one of our own
-annotations and not an unrelated string chunks[1] is annotation type
-(Calls/MustCall) chunks[2] is "target" keyword, chunks[3] is "=" chunks[4] is
-the target itself (FUNCTION/STRUCT potentially with sensible PARAM and FIELD
-specifiers) chunks[5] is "methods" keyword, chunks[6] is "=" chunks[7..end]
-are all the methods
+'free2']
+
+chunks[0] should be TOOL_CHECKER to help verify it's one of our own
```

Comment:
add "the literal string" and some quotes here to make it clear that `TOOL_CHECKER` is not intended to be a variable: `chunks[0] should be the literal string "TOOL_CHECKER" to help ...`

(Aside: should we choose a less-generic name than TOOL_CHECKER?)

Reply:
> should we choose a less-generic name than TOOL_CHECKER? 

How is `C_RESOURCE_LEAK_CHECKER_ANNOTATION`? Not sure what a good, less-generic name should be. Nonetheless, I think I'll be removing the 'TOOL_CHECKER' part entirely. I added it in the first place since how I first retrieved annotations was by looking at the global annotations generated in an LLVM IR file, and sometimes there would be unrelated strings. But since we now get the annotations by the AST and store them in a tmpfile, the 'TOOL_CHECKER' part is really not needed

Reply:
Removing it entirely sounds good to me.